test/prebuild.cu

```diff
@@ -70,57 +56,32 @@ __global__ void verify_correctness(const int *key, const int *col1, const int *c
  * contains 0,3,6,9...etc. The second column is filled with consecutive integers and is used as
  * payload column.
  */
-std::unique_ptr<table> generate_table(int multiple)
+std::unique_ptr<table> generate_table(cudf::size_type size, int multiple)
 {
   std::vector<std::unique_ptr<cudf::column>> new_table;
 
   // construct the key column
-  auto key_column = cudf::make_numeric_column(cudf::data_type(cudf::type_id::INT32), SIZE);
+  auto key_column = cudf::make_numeric_column(cudf::data_type(cudf::type_id::INT32), size);
   auto key_buffer = key_column->mutable_view().head<int>();
-  thrust::sequence(thrust::device, key_buffer, key_buffer + SIZE, 0, multiple);
+  thrust::sequence(thrust::device, key_buffer, key_buffer + size, 0, multiple);
   new_table.push_back(std::move(key_column));
 
   // construct the payload column
-  auto payload_column = cudf::make_numeric_column(cudf::data_type(cudf::type_id::INT32), SIZE);
+  auto payload_column = cudf::make_numeric_column(cudf::data_type(cudf::type_id::INT32), size);
   auto payload_buffer = payload_column->mutable_view().head<int>();
-  thrust::sequence(thrust::device, payload_buffer, payload_buffer + SIZE);
+  thrust::sequence(thrust::device, payload_buffer, payload_buffer + size);
   new_table.push_back(std::move(payload_column));
 
   return std::make_unique<table>(std::move(new_table));
 }
 
-int main(int argc, char *argv[])
+void run_test(cudf::size_type size,  // must be a multiple of 5
```

Comment:
Why this should be multiple of 5? Also, if it is a requirement, should we add `assert` or some error reported if it isn't? Maybe you have it in the code, I wasn't able to find it easily.

Reply:
Added the assert.

The size is a multiple of 5 so that we can test the output size easily in https://github.com/rapidsai/distributed-join/blob/e6c37662ed5d23087d6491a33563211cc2e4375b/test/prebuild.cu#L136.